openstack-cloud/src/main/java/jenkins/plugins/openstack/compute/JCloudsLauncher.java

```diff
@@ -43,7 +45,24 @@ public void launch(SlaveComputer computer, TaskListener listener) throws IOExcep
             throw new IOException("goto sleep");
         }
 
+        LOGGER.info("Launching host: " + host);
+
         if (slave.getSlaveType() == JCloudsCloud.SlaveType.SSH) {
+            // The SSHLauncher retry options do not seem to work reliably.
+            // As such, a configurable launch delay may be needed before
+            // attempting to launch over SSH.
+            JCloudsCloud cloud = JCloudsCloud.getByName(slave.getCloudName());
+            int sshLaunchDelay = cloud.getSshLaunchDelay();
+
+            if (sshLaunchDelay > 0) {
+                try {
+                    LOGGER.info(String.format("Delaying launch for %s milliseconds", sshLaunchDelay));
+                    Thread.sleep(sshLaunchDelay);
+                } catch (InterruptedException ex) {
+                    Thread.currentThread().interrupt();
```

Comment:
Is there any use-case when the launching thread is actually interrupted? If there is not, I recommend to propagate the exception up.